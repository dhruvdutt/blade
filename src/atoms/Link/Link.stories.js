import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, select } from '@storybook/addon-knobs';
import Link from './Link';

const sizeOptions = {
  xsmall: 'xsmall',
  small: 'small',
  medium: 'medium',
  large: 'large',
};

storiesOf('Link', module)
  .addParameters({
    component: Link,
  })
  .add('default', () => (
    <Link size={select('Size', sizeOptions, 'large')}>
      {text('Display Text', 'https://www.razorpay.com')}
    </Link>
  ))
  .add('default with href', () => (
    <Link size={select('Size', sizeOptions, 'large')} href="https://razorpay.com">
      {text('Display Text', 'https://www.razorpay.com')}
    </Link>
  ))
  .add('default with href and target', () => (
    <Link size={select('Size', sizeOptions, 'large')} hreg="https://razorpay.com" target="_blank">
      {text('Display Text', 'https://www.razorpay.com')}
    </Link>
  ));
