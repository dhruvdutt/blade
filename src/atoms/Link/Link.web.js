import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Text from '../Text';
import { getColor } from '../../_helpers/theme';
import automation from '../../_helpers/automation-attributes';

const StyledText = styled(Text)`
  color: ${(props) => getColor(props.theme, 'primary.800')};
  text-decoration: none;
  cursor: pointer;
  &:hover {
    text-decoration: underline;
    color: ${(props) => getColor(props.theme, 'primary.800')};
  }
  &:focus {
    outline: none;
    text-decoration: underline;
    color: ${(props) => getColor(props.theme, 'primary.900')};
  }
  &:active {
    outline: none;
    text-decoration: underline;
    color: ${(props) => getColor(props.theme, 'primary.700')};
  }
  &:visited {
    outline: none;
    color: ${(props) => getColor(props.theme, 'primary.700')};
  }
`;

const Link = ({ size, children, href, target, testId }) => {
  return (
    <StyledText as="a" size={size} href={href} target={target} {...automation(testId)}>
      {children}
    </StyledText>
  );
};

Link.propTypes = {
  children: PropTypes.string.isRequired,
  size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'large']),
  href: PropTypes.string,
  target: PropTypes.oneOf(['_blank', '_self']),
  testId: PropTypes.string,
};

Link.defaultProps = {
  size: 'medium',
  href: '#',
  target: '_self',
  testId: 'ds-text',
};

export default Link;
