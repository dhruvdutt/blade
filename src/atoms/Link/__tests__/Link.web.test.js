import React from 'react';
import Link from '../index';
import { renderWithTheme } from '../../../_helpers/testing';

describe('<Link />', () => {
  it('renders a link', () => {
    const displayText = 'Displaying some link';
    const { container } = renderWithTheme(<Link>{displayText}</Link>);
    expect(container).toMatchSnapshot();
  });

  it('renders a component by applying small', () => {
    const displayText = 'Displaying some link';
    const { container } = renderWithTheme(<Link size="small">{displayText}</Link>);
    expect(container).toMatchSnapshot();
  });

  it('renders a component by applying medium', () => {
    const displayText = 'Displaying some link';
    const { container } = renderWithTheme(<Link size="medium">{displayText}</Link>);
    expect(container).toMatchSnapshot();
  });

  it('renders a component by applying large', () => {
    const displayText = 'Displaying some link';
    const { container } = renderWithTheme(<Link size="large">{displayText}</Link>);
    expect(container).toMatchSnapshot();
  });
});
