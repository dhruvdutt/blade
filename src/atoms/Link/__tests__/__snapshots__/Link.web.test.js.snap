// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> Selectors onHover 1`] = `[MockFunction]`;

exports[`<Link /> Size large 1`] = `
<div>
  <a
    class="Textweb__Text-sc-6jf9e4-0 Linkweb__StyledText-sc-1u7mgfj-0 dgFRQe"
    color="shade.980"
    data-testid="ds-text"
    href="#"
    target="_self"
  >
    Displaying some link
  </a>
</div>
`;

exports[`<Link /> Size medium 1`] = `
<div>
  <a
    class="Textweb__Text-sc-6jf9e4-0 Linkweb__StyledText-sc-1u7mgfj-0 eDaeyA"
    color="shade.980"
    data-testid="ds-text"
    href="#"
    target="_self"
  >
    Displaying some link
  </a>
</div>
`;

exports[`<Link /> Size small 1`] = `
<div>
  <a
    class="Textweb__Text-sc-6jf9e4-0 Linkweb__StyledText-sc-1u7mgfj-0 dzuwNk"
    color="shade.980"
    data-testid="ds-text"
    href="#"
    target="_self"
  >
    Displaying some link
  </a>
</div>
`;

exports[`<Link /> href href 1`] = `
<div>
  <a
    class="Textweb__Text-sc-6jf9e4-0 Linkweb__StyledText-sc-1u7mgfj-0 dgFRQe"
    color="shade.980"
    data-testid="ds-text"
    href="https://www.razorpay.com"
    target="_self"
  >
    Displaying some link
  </a>
</div>
`;

exports[`<Link /> href target 1`] = `
<div>
  <a
    class="Textweb__Text-sc-6jf9e4-0 Linkweb__StyledText-sc-1u7mgfj-0 dgFRQe"
    color="shade.980"
    data-testid="ds-text"
    href="https://www.razorpay.com"
    target="_blank"
  >
    Displaying some link
  </a>
</div>
`;

exports[`<Link /> renders a link 1`] = `
<div>
  <a
    class="Textweb__Text-sc-6jf9e4-0 Linkweb__StyledText-sc-1u7mgfj-0 eDaeyA"
    color="shade.980"
    data-testid="ds-text"
    href="#"
    target="_self"
  >
    Displaying some link
  </a>
</div>
`;
